src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/ClientManager.java

```diff
@@ -0,0 +1,122 @@
+import com.snowflake.kafka.connector.internal.SnowflakeErrors;
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+/** This is a wrapper to help manage the streaming ingest clients */
+public class ClientManager {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
+
+  protected ClientManager() {
+    LOGGER = new LoggerHandler(this.getClass().getName());
+    this.streamingIngestClientCount = 0;
+  }
+
+  // ONLY FOR TESTING - use this to inject client
+  @VisibleForTesting
+  public ClientManager(SnowflakeStreamingIngestClient client) {
+    this();
+    this.streamingIngestClient = client;
+  }
+
+  /**
+   * Calls the ingest sdk to create the streaming client
+   *
+   * @param connectorConfig properties for the streaming client
+   * @param kcInstanceId identifier of the connector instance creating the client
+   */
+  public void createStreamingClient(Map<String, String> connectorConfig, String kcInstanceId) {
+    Map<String, String> streamingPropertiesMap =
+        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig));
+    Properties streamingClientProps = new Properties();
+    streamingClientProps.putAll(streamingPropertiesMap);
+
+    String streamingIngestClientName = this.getStreamingIngestClientName(kcInstanceId);
+
+    try {
+      LOGGER.info("Creating Streaming Client. ClientName:{}", streamingIngestClientName);
+      this.streamingIngestClientCount++;
+      this.streamingIngestClient =
+          SnowflakeStreamingIngestClientFactory.builder(streamingIngestClientName)
+              .setProperties(streamingClientProps)
+              .build();
+    } catch (SFException ex) {
+      LOGGER.error(
+          "Exception creating streamingIngestClient with name:{}", streamingIngestClientName);
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Calls the ingest sdk to close the client sdk
+   *
+   * @return true if the client was successfully closed, false if not
+   */
+  public boolean closeStreamingClient() {
+    if (this.streamingIngestClient == null || this.streamingIngestClient.isClosed()) {
+      LOGGER.info("Streaming client is already closed or null");
+      return true;
+    }
+
+    String streamingIngestClientName = this.streamingIngestClient.getName();
+    LOGGER.info("Closing Streaming Client:{}", streamingIngestClientName);
+
+    try {
+      this.streamingIngestClient.close();
+      return true;
```

Comment:
Nit, you could remove the `return true` outside of the try/catch

Reply:
Apologies I dont remember the context for this code, and cant find the old version on this comment? Resolving because I don't think it applies anymore? Reraise if it does